launcher/src/main/java/org/wso2/carbon/launcher/CarbonServer.java

```diff
@@ -278,6 +284,7 @@ public ServerStatus getServerCurrentStatus() {
      */
     private void dispatchEvent(int event) {
         CarbonServerEvent carbonServerEvent = new CarbonServerEvent(event, config);
+        carbonServerEvent.setSystemBundleContext(framework.getBundleContext());
```

Comment:
Hi @imesh , This sets the system bundle context for every carbon event, so that any listener can access the system bundle if required.